### PR TITLE
Cleanup source repository, create dist zip (#1305)

### PR DESCRIPTION
**Pull request by jekkos**
Planning to merge my cleanup branch as soon as someone tests the latest bintray zip

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -20,8 +20,7 @@
 		"php": "^5.5 || ^7.0",
 		"codeigniter/framework": "^3.1.2",
 		"ins0/google-measurement-php-client": "dev-master#2b3aef2375837339f5b2bfe88e76994481f86264",
-		"dompdf/dompdf": "^0.7.0",
-		"tamtamchik/namecase": "^1.0"
```
Comment (jekkos) [inline composer.json]: where did we need this one exactly?

Comment (daN4cat) [inline composer.json]: it's in the code and fixes people's name capitalisation when saved. This library does the proper job even with the names with de den van and etc. Tested.

**Comment by daN4cat** [comment]
Why have you removed the public files? Now if I pull from Github I get broken stuff. .....

**Comment by jekkos** [comment]
You mean the minified files?  Because they don't  belong in a sourcecode repo?

**Comment by jekkos** [comment]
@daN4cat any thoughts then on this one? I don't think it makes sense having to merge minified files all the time... they always crash my merge editor..
of course it's less convenient but that's how it's done in most projects, no?

**Comment by daN4cat** [comment]
let me give a try....

**Comment by odiea** [comment]
I lost the download link for the newly created zip.

**Comment by jekkos** [comment]
Ok to merge then?

**Comment by daN4cat** [comment]
@jekkos working on it and made last round of fixes (hopefully)!

**Comment by daN4cat** [comment]
@jekkos any idea how we used to update compoer.LICENSES file? It doesn't look like there is any script or similar to do that. I updated it manually.

**Comment by jekkos** [comment]
Think it's a builtin composer command? Didn't we have it setup in grunt?

**Comment by daN4cat** [comment]
No

**Comment by jekkos** [comment]
I might have done it manually at the time then, there's composer licenses --format=json

**Comment by daN4cat** [comment]
is public/dist supposed to be completely clean in Github?

**Comment by jekkos** [comment]
Yes these are all files copied by the build

**Comment by daN4cat** [comment]
OK, going to commit that tidy up too... a lot of stuff was left behind

**Comment by jekkos** [comment]
OK great that'll make the source repo a bit lighter already

**Comment by jekkos** [comment]
I'll pick the latest zip and see if it works, also I'll adapt README.md with link to the zips

**Comment by daN4cat** [comment]
I assume even bower.LICENSES is generated right?

**Comment by daN4cat** [comment]
I would need to create a script to regenerate the complete development environment.

**Comment by jekkos** [comment]
Ok first npm install, then bower install, and that should be it, no?

**Comment by daN4cat** [comment]
composer...

**Comment by jekkos** [comment]
Ok I've did some change now I append the git revision to the zipfile. Not sure how bintray will list the files (hopefully newest first). Other option would be to add a timestamp to differentiate different builds

**Comment by daN4cat** [comment]
In the deployment step have you installed CI translations?

**Comment by daN4cat** [comment]
@jekkos something is broken with travis now, since the script change you made

**Comment by daN4cat** [comment]
Just checking bintray zip file:
- /public/bower_components, is probably not necessary. I would say no unless somebody is after the debug version
- /public/images/menubar/png/ can be removed as it contains load of images
- /public/dist/bootswatch/ subdir contains more than the min.css, those files are not necessary (be careful that fonts need to stay as it is
- /vendor/codeigniter/framework/system/language/ is missing all the languages

**Comment by daN4cat** [comment]
It's annoying the composer packages cannot be deployed cleanly without ton of test and example stuff. There should be a means of getting a clean deployable package.

**Comment by daN4cat** [comment]
BTW bintray file is old and dated 03.08.17, so something is not completely automated yet.

**Comment by jekkos** [comment]
Yes it's not refreshing the file, I'll give it another try

**Comment by daN4cat** [comment]
Also travis script is throwing an error

File: application/views/errors/html/error_general_LOCAL_10171.php [hunk 2]
```diff
@@ -0,0 +1,64 @@
+<?php
```
Comment (daN4cat) [inline application/views/errors/html/error_general_LOCAL_10171.php]: what's this?

**Comment by jekkos** [comment]
Ok so I need to add CI language install step in travis as well

**Comment by daN4cat** [comment]
You need to run: `php bin/install.php translations develop`

**Comment by jekkos** [comment]
Ok added that part

**Comment by daN4cat** [comment]
Update the README with the bintray link.

I bet that Steve and odiea will complaint about this change...

**Comment by odiea** [comment]
Once it's ready to play with I will let you know. So far you guys have done a great job spending your free time on this project. Thanks

**Comment by jekkos** [comment]
OK all done. Zip files can be found at

https://bintray.com/jekkos/opensourcepos/opensourcepos/3.1.0#files

**Comment by odiea** [comment]
What is and where does it go?
opensourcepos.20170805225340.zip.asc

**Comment by jekkos** [comment]
ah that just a checksum.. shows that the file has not been tampered with.

**Comment by odiea** [comment]
This download fired right up with no issues so far.

**Pull request merged**